### PR TITLE
SOAPUIOS-608 Swagger in SoapUI

### PR DESCRIPTION
**Pull request by AlexanderNikitin-Smartbear**
https://smartbear.atlassian.net/browse/SOAPUIOS-608

### CODE REVIEW

File: soapui/src/main/java/com/eviware/soapui/impl/support/MediaTypeUtils.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package com.eviware.soapui.impl.support;
+
+import javax.ws.rs.core.MediaType;
+
+public class MediaTypeUtils {
```
Comment (AlexanderYeremeyev) [inline soapui/src/main/java/com/eviware/soapui/impl/support/MediaTypeUtils.java:5]: first question was about javadoc.
But it is a copy of existing code that is why it is possible implementation.
But absolutely unclear :(

File: soapui/src/main/java/com/smartbear/swagger/ExportSwaggerAction.java [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+    public static final String YAML_EXTENSION = ".yaml";
+
+    private static final String BASE_PATH = Form.class.getName() + Form.BASEPATH;
+    private static final String TARGET_PATH = Form.class.getName() + Form.FOLDER;
+    private static final String FORMAT = Form.class.getName() + Form.FORMAT;
+    private static final String VERSION = Form.class.getName() + Form.VERSION;
+    private static final String SWAGGER_VERSION = Form.class.getName() + Form.SWAGGER_VERSION;
+
+    private static final String SWAGGER_2_0 = "Swagger 2.0";
+    private static final String OPEN_API_3_0 = "OpenAPI 3.0";
```
Comment (AlexanderYeremeyev) [inline soapui/src/main/java/com/smartbear/swagger/ExportSwaggerAction.java:42]: as i remember we decided to skip it.

File: soapui/src/main/java/com/smartbear/swagger/ExportSwaggerAction.java [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+                    UISupport.showInfoMessage(String.format(messages.get("ExportSwaggerAction.InfoMessage.ListingHasBeenCreated"), path));
+                }
+
+                settings.setString(BASE_PATH, dialog.getValue(Form.BASEPATH));
+                settings.setString(TARGET_PATH, dialog.getValue(Form.FOLDER));
+                settings.setString(FORMAT, dialog.getValue(Form.FORMAT));
+                settings.setString(VERSION, dialog.getValue(Form.VERSION));
+                settings.setString(SWAGGER_VERSION, dialog.getValue(Form.SWAGGER_VERSION));
+
+                Analytics.trackAction("ExportSwagger",
```
Comment (AlexanderYeremeyev) [inline soapui/src/main/java/com/smartbear/swagger/ExportSwaggerAction.java:128]: 👍

File: soapui/src/main/java/com/smartbear/swagger/Swagger2Importer.java [hunk 4]
```diff
@@ -0,0 +1,496 @@
...
+                        Model modelDefinition = swagger.getDefinitions().get(refModel.getSimpleRef());
+                        if (modelDefinition instanceof ComposedModel) {
+                            objectProperty = null;
+                        } else if (modelDefinition != null) {
+                            objectProperty = new ObjectProperty(modelDefinition.getProperties());
+                            objectProperty.name(refModel.getSimpleRef());
+                        }
+                    }
+                    Example output = objectProperty != null ? ExampleBuilder.fromProperty(objectProperty, swagger.getDefinitions()) :
+                            ExampleBuilder.fromModel(null, bodyParameterModel, swagger.getDefinitions(), new HashSet<String>());
```
Comment (Alexander-Paeshin) [inline soapui/src/main/java/com/smartbear/swagger/Swagger2Importer.java:314]: The example builder shouldn't be used in SoapUI OS

File: soapui/src/main/java/com/smartbear/swagger/SwaggerImporter.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
+package com.smartbear.swagger;
+
+import com.eviware.soapui.impl.rest.RestService;
+
+public interface SwaggerImporter {
+
+    String AUTHORIZATION_HEADER = "Authorization";
```
Comment (AlexanderYeremeyev) [inline soapui/src/main/java/com/smartbear/swagger/SwaggerImporter.java:7]: not the best idea to store constant inside the interface.

File: soapui/src/main/java/com/smartbear/swagger/utils/OpenAPIUtils.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+import io.swagger.util.Json;
+import io.swagger.util.Yaml;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class OpenAPIUtils {
+
+    private static final Logger LOG = LoggerFactory.getLogger(OpenAPIUtils.class);
+
+    public static String extractStringFromExampleObject(String contentType, Object example) {
```
Comment (Alexander-Paeshin) [inline soapui/src/main/java/com/smartbear/swagger/utils/OpenAPIUtils.java:15]: This is also part of preparing sample for request and, as I see, it is not used anywhere.

File: pom.xml [hunk 7]
```diff
@@ -137,6 +137,14 @@
         <fest-swing.version>1.2.1</fest-swing.version>
         <maven-checksum-plugin.version>1.0</maven-checksum-plugin.version>
         <jackson-version>2.13.0</jackson-version>
+
+        <!-- Swagger/OpenAPI -->
+
+        <io.swagger.core.v3.swagger-models.version>2.1.7</io.swagger.core.v3.swagger-models.version>
```
Comment (AlexanderYeremeyev) [inline pom.xml]: can we remove these empty values?

**Review by AlexanderYeremeyev — CHANGES_REQUESTED** [review]

File: soapui/src/main/java/com/smartbear/swagger/ExportSwaggerAction.java [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+                SwaggerExporter exporter = null;
+                String target = null;
+
+                if (swaggerVersion.equals(SWAGGER_2_0)) {
+                    exporter = new Swagger2Exporter(project);
+                    target = dialog.getValue(Form.FOLDER);
+                }
+
+                //temp condition
+                if(exporter == null) {
```
Comment (AlexanderYeremeyev) [inline soapui/src/main/java/com/smartbear/swagger/ExportSwaggerAction.java:109]: 🥇

File: soapui/src/main/java/com/smartbear/swagger/ExportSwaggerAction.java [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+                    return;
+                }
+
+                String path = exporter.exportToFileSystem(target, version, format, services, dialog.getValue(Form.BASEPATH));
+
+                if (path == null) {
+                    UISupport.showInfoMessage(messages.get("ExportSwaggerAction.InfoMessage.NothingCreated"));
```
Comment (AlexanderYeremeyev) [inline soapui/src/main/java/com/smartbear/swagger/ExportSwaggerAction.java]: bundles

**Review by AlexanderYeremeyev — APPROVED** [review]
Export will work.
Exactly for me import doesn't work. But @Alexander-Paeshin said that it work for him.
Approve this PR.

File: soapui/src/main/java/com/eviware/soapui/impl/support/MediaTypeUtils.java (hunk 1 above)
Comment (AlexanderNikitin-Smartbear) [reply soapui/src/main/java/com/eviware/soapui/impl/support/MediaTypeUtils.java:5]: 1. Copied from ReadyAPI. There is no javadoc initially.
2. What kind of javadoc you want to the MediaTypeUtils class? Are you sure that common description for the class like "Some utility methods" says to anybody more than class name?

File: soapui/src/main/java/com/smartbear/swagger/SwaggerImporter.java (hunk 5 above)
Comment (AlexanderNikitin-Smartbear) [reply soapui/src/main/java/com/smartbear/swagger/SwaggerImporter.java:7]: Initially the field was in the AbstactSwaggerImporter wich is not needed.

**Pull request merged**